scripts/lighthouse.js

```diff
@@ -1,33 +1,82 @@
-const server = require('../lib/server')
-
+const { exec } = require('child_process')
+const { writeFileSync } = require('fs')
 const lighthouse = require('lighthouse')
-const chromeLauncher = require('chrome-launcher')
-
-function launchChromeAndRunLighthouse (url, opts, config = null) {
-  return chromeLauncher.launch({chromeFlags: opts.chromeFlags}).then(chrome => {
-    opts.port = chrome.port
-    return lighthouse(url, opts, config).then(results => {
-      // The gathered artifacts are typically removed as they can be quite large (~50MB+)
-      delete results.artifacts
-      return chrome.kill().then(() => results)
-    })
-  })
+const { resolve } = require('path')
+const puppeteer = require('puppeteer')
+const { parse: parseUrl } = require('url')
+const { promisify } = require('util')
+const waitOn = promisify(require('wait-on'))
+const createBadge = require('gh-badges')
+
+function unhandledRejectionHandler(error) {
+  console.error(error)
+  process.exit(1)
 }
 
-const opts = {
-  chromeFlags: ['--show-paint-rects']
+function badgeColor(score) {
+  if (score > 90) return 'green'
+  else if (score > 80) return 'yellow'
+  else if (score > 50) return 'orange'
+
+  return 'red'
 }
 
-server.init()
-  .then(({ app, server }) => {
-    console.log('\n\n server started on port 3000')
-    launchChromeAndRunLighthouse('http://localhost:3000/', opts).then(results => {
-      console.log(results)
-      server.close()
-      process.exit()
+async function badge(spec) {
+  return new Promise((resolve, reject) => {
+    createBadge(spec, (svg, err) => {
+      if (err) return reject(err)
+
+      resolve(svg)
     })
   })
-  .catch(err => {
-    console.error(err)
-    process.exit(1)
-  })
+}
+
+async function main() {
+  const url = process.argv[2] || 'http://localhost:3000'
+  const parsedUrl = parseUrl(url)
+
+  // Launch the server and wait for the port to be available
+  let server = null
+  if (!process.argv[2]) {
+    // No external host, start the server
+    server = exec('npm run start')
+    await waitOn({ resources: [`tcp:${parsedUrl.port}`] })
+  }
+
+  // Launch chrome
+  const browser = await puppeteer.launch({ headless: process.env.VIEW !== 'true', args: ['--no-sandbox', '--show-paint-rects'] })
+
+  // Execute lighthouse
+  const results = (await lighthouse(url, { port: parseUrl(browser.wsEndpoint()).port, output: 'json' })).lhr
+  delete results.report
+  delete results.artifacts
+
+  // Kill everything
+  await browser.close()
+  if (server) server.kill('SIGTERM')
+
+  // Gather results and build badges
+  const outputItems = []
+
+  for (const category of Object.values(results.categories)) {
+    const score = Math.ceil(category.score * 100)
+    outputItems.push(`${category.title}: ${score}`)
+
+    const svg = await badge({
+      text: [`Ligthouse ${category.title} Score`, `${score}/100 `],
+      colorscheme: badgeColor(score),
+      template: 'flat'
+    })
```

Comment:
It would be nice to actually generate these badges in parallel instead of waiting on each other.

The code could look like:

```js
const promises = []
for (const category of ...) {
  // Do stuff
  promises.push(badge({ ... }))
}
(await Promise.all(promises)).forEach((svg) => ...)
```